contracts/collections/sg721-base/src/contract.rs

```diff
@@ -335,6 +372,20 @@ where
             royalty_info: royalty_info_res,
         })
     }
+
+    pub fn migrate(deps: DepsMut, _env: Env, _msg: Empty) -> Result<Response, ContractError> {
+        // make sure the correct contract is being upgraded, and it's being
+        // upgraded from the correct version.
+        let version = cw2::get_contract_version(deps.storage)?;
+        cw2::assert_contract_version(deps.as_ref().storage, CONTRACT_NAME, EXPECTED_FROM_VERSION)
```

Comment:
I think checking the exact previous version is excessive because it adds more work for every migration and only allows migrating from the previous version to the next. Just making sure the new version is greater than the old version is fine imho.

Reply:
Copying logic from cw-nfts: https://github.com/CosmWasm/cw-nfts/pull/112/files#diff-733b03d19162a7149fd3e3feb6552118ea6a667aac4cbcda9ece681ad6425ef7R83

Reply:
@yubrew are you asking to change something in the code here? Not sure what the action to take is.

Reply:
> Just making sure the new version is greater than the old version is fine imho.

@shanev If the migration from version `A` to version `B` includes some state changes, and from `B` to `C` also includes some state changes, then you can't directly apply the version `C` migration method on a contract that's still on `A`.